### PR TITLE
Add maven-publish gradle plugin

### PR DESCRIPTION
**Pull request by trask**
Enable publishing of artifacts to local maven repository so they can be used to test vendor-specific agent based on auto instrumentation artifacts.

### CODE REVIEW

File: trace-api/trace-api.gradle [hunk 1]
```diff
@@ -1,3 +1,7 @@
+plugins {
+  id 'maven-publish'
```
Comment (tylerbenson) [inline trace-api/trace-api.gradle]: Can we avoid publishing this jar for now?

File: instrumentation/instrumentation.gradle [hunk 2]
```diff
@@ -70,6 +70,30 @@ subprojects { Project subProj ->
...
+    apply plugin: 'maven-publish'
+    publishing {
+      publications {
+        mavenJava(MavenPublication) {
+          from components.java
+          afterEvaluate {
+            def parts = project.path.split(':')
+            if (parts.size() == 3) {
+              groupId = 'io.opentelemetry.auto.instrumentation'
+              artifactId = "instrumentation-$project.name"
```
Comment (tylerbenson) [inline instrumentation/instrumentation.gradle]: We may want to separate the three different types here somehow:  decorators/helpers, auto instrumentation, and manual instrumentation.

File: trace-api/trace-api.gradle (hunk 1 above)
Comment (trask) [reply trace-api/trace-api.gradle]: `trace-api` is now completely removed!

File: instrumentation/instrumentation.gradle (hunk 2 above)
Comment (trask) [reply instrumentation/instrumentation.gradle]: I'm thinking let's not publish artifacts to maven central for v0.1.0 to give us more time to figure out artifact naming. For v0.1.0 we can just publish the agent jar to github release.

Comment (tylerbenson) [reply instrumentation/instrumentation.gradle]: I'm fine with that.  Does that mean close this PR?

Comment (trask) [reply instrumentation/instrumentation.gradle]: Yeah, that's probably safest

**Pull request closed without merging**